lib/buffer.js

```diff
@@ -289,6 +289,24 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
+Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+  if (byteOffset > 0xffffffff)
+    byteOffset = 0xffffffff;
+  if (byteOffset < 0)
+    byteOffset = 0;
+  byteOffset = byteOffset >>> 0;
+
+  if (typeof val === 'string')
+    return internal.indexOfString(this, val, byteOffset);
+  if (val instanceof Buffer)
+    return internal.indexOfBuffer(this, val, byteOffset);
+  if (typeof val === 'number')
+    return internal.indexOfNumber(this, val, byteOffset);
```

Comment:
FWIW I've seen ~5% speed increase in speed by not using the type checking functions in `util` when [optimizing the `url` module](https://github.com/iojs/io.js/pull/419#issuecomment-70035382).